src/txn.h

```diff
@@ -31,26 +36,30 @@ typedef enum {
 	TXN_ELEM_TXN_SIG,
 } txnElemType_e;
 
+typedef struct {
+    txnElemType_e elemType; // type of most-recently-seen element
```

Comment:
IIRC enums are the same size as `int`, but we only need 1 byte here. Looks like some compilers have an `-fshort-enums` flag; if that doesn't work, I suppose we can make this type `uint8_t` and document its real type in the comment.